### ISSUE TITLE
Fallback to cargo build if prebuilt module is not available

### ISSUE THREAD

**Issue by timfish** [napi-rs/napi-rs#447]
Would it be worth having an `napi postinstall` command that would display a warning and call `cargo build` if no pre-built binary can be found?

**Comment by Brooooooklyn** [comment]
Yes I think it`s worth to do it, especially for  platforms which are hard to prebuild in CI, such as freebsd.

**Comment by timfish** [comment]
Maybe add it as a flag on `build`?
```json
  "scripts": {
    "postinstall": "napi build --release --skip-if-binary-found",
  }
```

**Comment by Brooooooklyn** [comment]
Looks reasonable, but I prefer to determine if it need to build from source by detect the `npm_config_build_from_source ` environment variable, it will be presented when `npm install --build-from-source`

**Comment by Brooooooklyn** [comment]
I've done this in https://github.com/Brooooooklyn/pinyin/blob/main/postinstall.js

**Comment by jsjoeio** [comment]
Does `napi-rs` support `process.env.npm_config_build_from_source` currently? I couldn't find it in the source. I think we're running into this issue in code-server (https://github.com/coder/code-server/issues/4804)

**Comment by Brooooooklyn** [comment]
@jsjoeio Sorry it took so long to reply to you, I just got back to work from my vacation.

Unfortunately `process.env.npm_config_build_from_source` is not supported, because I think it's not easy for users to install Rust and related toolchains. And build Rust from the source is very time-consuming.

But `yarn 1.x` has a workaround for this scenario, you can use `--ignore-platform` flag to let `yarn` install all the platform binaries for you. And `yarn 2+` here is [supportedArchitectures](https://yarnpkg.com/configuration/yarnrc#supportedArchitectures) for this case.

**Comment by jsjoeio** [comment]
No worries at all! Hope it was a nice vacation :D 

> But yarn 1.x has a workaround for this scenario, you can use --ignore-platform flag to let yarn install all the platform binaries for you. And yarn 2+ here is [supportedArchitectures](https://yarnpkg.com/configuration/yarnrc#supportedArchitectures) for this case.

Ah, I see. Good to know. I'll share with the code-server team.
